### PR TITLE
Align the server session handle and engine session handle for Spark engine

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Align the server session handle and engine session handle for Spark engine.

It make it easy to recovery the engine session in any kyuubi instance easy.

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSessionImpl.scala [hunk 1]
```diff
@@ -39,6 +40,10 @@ class SparkSessionImpl(
     val spark: SparkSession)
   extends AbstractSession(protocol, user, password, ipAddress, conf, sessionManager) {
 
+  override val handle: SessionHandle =
+    conf.get(KyuubiReservedKeys.KYUUBI_SESSION_HANDLE_KEY).map(SessionHandle.fromUUID).getOrElse(
+      SessionHandle())
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSessionImpl.scala:45]: `KYUUBI_SESSION_HANDLE_KEY` should always present
```suggestion
  override val handle: SessionHandle = SessionHandle.fromUUID(conf(KYUUBI_SESSION_HANDLE_KEY))
```

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala [hunk 2]
```diff
@@ -135,21 +136,24 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
       password: String,
       ipAddress: String,
       conf: Map[String, String]): Session = {
-    val sparkSession =
-      try {
-        getOrNewSparkSession(user)
-      } catch {
-        case e: Exception => throw KyuubiSQLException(e)
-      }
+    conf.get(KyuubiReservedKeys.KYUUBI_SESSION_HANDLE_KEY).map(SessionHandle.fromUUID).flatMap(
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:139]: ditto

**Comment by turboFei** [comment]
addressed @pan3793

**Review by pan3793 — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to master and 1.7

**Pull request closed without merging**